diff --git a/node_modules/expo-head/build/ExpoHead.ios.js b/node_modules/expo-head/build/ExpoHead.ios.js
index b29c6e6..599c0e7 100644
--- a/node_modules/expo-head/build/ExpoHead.ios.js
+++ b/node_modules/expo-head/build/ExpoHead.ios.js
@@ -165,7 +165,7 @@ const activities = new Map();
 function useRegisterCurrentActivity(activity) {
     // ID is tied to Expo Router and agnostic of URLs to ensure dynamic parameters are not considered.
     // Using all segments ensures that cascading routes are considered.
-    const activityId = urlToId(usePathname());
+    const activityId = urlToId(usePathname() ?? '/') ?? 'backup';
     const cascadingId = urlToId(useSegments().join("-") || "-");
     const activityIds = Array.from(activities.keys());
     const cascadingActivity = React.useMemo(() => {
@@ -195,7 +195,8 @@ function useRegisterCurrentActivity(activity) {
         }
         previousActivity.current = cascadingActivity;
         if (!cascadingActivity.id) {
-            throw new Error("Activity must have an ID");
+            cascadingActivity.id = urlToId(cascadingActivity.webpageURL); 
+            // throw new Error("Activity must have an ID");
         }
         // If no features are enabled, then skip registering the activity
         if (cascadingActivity.isEligibleForHandoff ||
