diff --git a/node_modules/expo-router/src/static/renderStaticContent.tsx b/node_modules/expo-router/src/static/renderStaticContent.tsx
index 7fdae39..a64f081 100644
--- a/node_modules/expo-router/src/static/renderStaticContent.tsx
+++ b/node_modules/expo-router/src/static/renderStaticContent.tsx
@@ -11,7 +11,7 @@ import Head from "expo-router/head";
 import React from "react";
 import ReactDOMServer from "react-dom/server";
 import { AppRegistry } from "react-native-web";
-
+import * as Font from 'expo-font/build/server';
 import { getRootComponent } from "./getRootComponent";
 
 AppRegistry.registerComponent("App", () => App);
@@ -72,7 +72,7 @@ export function getStaticContent(location: URL): string {
   let output = mixHeadComponentsWithStaticResults(headContext.helmet, html);
 
   output = output.replace("</head>", `${css}</head>`);
-
+  output = output.replace('</head>', `${Font.getStaticResources().join('')}</head>`);
   return "<!DOCTYPE html>" + output;
 }
 
